src/lib/utils/wrap.js

```diff
@@ -0,0 +1,11 @@
+export default function wrap(Type, fn, dontCloneWithNoArgs) {
```

Comment:
With regard to moment-timezone - it calls update on the moment, and there's no need to be cloning the moment to do that - that's just a perf killer. Can we track updateInProgress in this function and just mutate if that's the case? I think that's fine to do because JavaScript is single threaded (thank god!).